packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -122,9 +141,43 @@ export class SchemaConverter implements SimpleDbSchemaConverter {
       p = p.next(() => this.ensureSequenceNumbers(txn));
     }
 
+    assert(
+      fromVersion <= 8,
+      'After version 8, we switch to a separately maintained schema ' +
+        'version. IndexedDB should never see its version go past 8.'
+    );
+    if (fromVersion < 8 && toVersion >= 8) {
+      p = p.next(() => addManualSchemaVersion(txn, db));
+    }
+
+    // IMPORTANT: new migrations should be implemented in
+    // `upgradeFromManualVersion`. After this point, we no longer use
+    // IndexedDB's versioning, as it does not allow downgrades.
+
     return p;
   }
 
+  doManualMigrations(
+    db: IDBDatabase,
+    toVersion: number
+  ): PersistencePromise<void> {
+    return readManualSchemaVersion(db).next(fromVersion => {
```

Comment:
1. I'm having trouble reasoning about our transaction usage here.  readManualSchemaVersion creates a new transaction, does some stuff, and then in this next() callback (so we're still inside that transaction?), we create _another_ transaction.  Does that result in "nested" transactions?  Or does opening a new transaction end the old one?  If so, should we be doing something with the completionPromise on the old one?  This would be cleaner / safer if we just wrapped `db` in a `SimpleDB` instance early on, and then used db.runTransaction() rather than manually instantiating SimpleDbTransaction (since runTransaction returns a Promise that resolves after the transaction is committed).

2. Why are we creating two transactions anyway?  I think we should be reading the current schema version and doing the migrations atomically.  Else, technically, two tabs could end up doing the migrations on top of each other or something (maybe non-harmful since we made them idempotent, but still...).